libc/src/string/strcmp.c

```diff
@@ -0,0 +1,10 @@
+#include <string.h>
+
+int strcmp(const char *str1, const char *str2){
```

Comment:
This function can be implemented a bit more efficiently. 

- You can just increment the pointers instead of creating an indexing variable. 
- The check for a null terminator on the second string is unnecessary, because of the comparison between the strings.

You also shouldn't cast away const on c1 and c2, the const applies to the value at the pointer. You can just use the str1 and str2 pointers instead.

Reply:
The pointers are constants, so a indexing variable is required. The check for null terminator on second string is required for correct return value (probably more things as well).

Reply:
The function is fixed

Reply:
> The pointers are constants, so a indexing variable is required. The check for null terminator on second string is required for correct return value (probably more things as well).

The pointers themself aren't const it's the values they are pointing to that are, you don't need and indexing variable.

What do you mean the second null terminator check is required for correct return value? I actually think this function is incorrect, it returns 0 if one of the strings terminate before the other one.

Reply:
Good catch!